### PR TITLE
minor cleanups

### PR DESCRIPTION
**Pull request by pq**
* update typedef syntax
* convert to `=>` short-hand

/cc @devoncarew @kevmoo

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 .pub
 packages
 pubspec.lock
+.idea
```
Comment (kevmoo) [inline .gitignore]: Please, no.  We should have folks ignore these globally!

File: lib/logging.dart [hunk 2]
```diff
@@ -247,7 +245,7 @@ class Logger {
 }
 
 /// Handler callback to process log entries as they are added to a [Logger].
-typedef void LoggerHandler(LogRecord record);
+typedef LoggerHandler = void Function(LogRecord record);
```
Comment (kevmoo) [inline lib/logging.dart]: I don't think this is used anywhere in the package. We should deprecate this, too!

Comment (devoncarew) [reply lib/logging.dart]: I don't see any uses internally...

File: .gitignore (hunk 1 above)
Comment (pq) [reply .gitignore]: 👍

File: lib/logging.dart (hunk 2 above)
Comment (pq) [reply lib/logging.dart]: Ah, and see also: https://github.com/dart-lang/logging/issues/13.

**Comment by pq** [comment]
Feedback addressed.  Thanks!

**Comment by kevmoo** [comment]
Make sure to update the commit message 😄

**Pull request merged**